select.go

```diff
@@ -1,7 +1,14 @@
 package ykoath
```

Comment:
I think the changes to `select.go` are here by mistake? At least the KDF parts?

Reply:
Most of this is heavily based on what I saw in `ykman`, upon implementation, the `DeviceID()` function is not actually used.

Here's an example of using `Validate()`: https://git.iamthefij.com/iamthefij/yk-cli/src/branch/master/main.go#L23

The validation process is a couple steps. First is to do a `Select()`, then to use your password to derive a `key`. That key is used in the `Validate()` call.

This is what the flow looks like (I'll put in readme or docstrings)
```
s, err := oath.Select()
passKey := s.DeriveKey(password)
ok, err := oath.Validate(s, passKey)
```

Reply:
I am not 100% sure I understand the scenario. Is the KDF selection schema is something application specific? Or is this part of the Yubikey protocol for password management? Or is it just the way `ykman` handles this?

Reply:
You mean the `Hash()` method where it returns different hash functions?

None of this is really application specific. This is how you get password protected content off the Yubikey. You have to implement a challenge response, which is handled in the `Validate()` function with the results of the `Select()` function.

I thought about actually wrapping this all up into something like `func (s Select) Validate(password string) error {}`, but this being split makes it a little easier to store the `passKey` rather than the `password` itself.